### PR TITLE
[Interval] Patch the need recomputing

### PR DESCRIPTION
**Pull request by RedGl0w**
This fixes https://github.com/numworks/epsilon/issues/666

### CODE REVIEW

**Comment by Ecco** [comment]
Hi @RedGl0w ! Thanks for this :) I don't know if this is the right fix, but it's so short I would assume it is. On the PR itself, just a few notes:
* Please put the setter directly in the header file.
* The getter should start with a lowercase letter
* It'd be better if the setter/getter matched the instance variable's name. So `bool needCompute()` and `void setNeedCompute(bool)`

**Comment by RedGl0w** [comment]
>     * The getter should start with a lowercase letter

It seems I won't ever manage to respect camelCase :sob:

File: apps/shared/interval.h [hunk 1]
```diff
@@ -18,6 +18,8 @@ class Interval {
   void setEnd(double f);
   void setStep(double f);
   void setElement(int i, double f);
+  bool needCompute() const { return m_needCompute; }
```
Comment (EmilieNumworks) [inline apps/shared/interval.h:21]: This does't need to be exposed, does it?

Comment (RedGl0w) [reply apps/shared/interval.h:21]: We don't need it, you're right.
I always add setter and getter, even when they're useless

File: apps/shared/interval.h [hunk 2]
```diff
@@ -18,6 +18,7 @@ class Interval {
   void setEnd(double f);
   void setStep(double f);
   void setElement(int i, double f);
+  void setNeedCompute(bool compute){ m_needCompute = compute;}
```
Comment (EmilieNumworks) [inline apps/shared/interval.h:21]: m_needCompute is never set to false through external code, only Interval::computeElements private method will turn m_needCompute to false. So the public API of Interval should only expose a method that can turn m_needCompute to true like `forceComputing()` - you might think of a better name!

Comment (RedGl0w) [reply apps/shared/interval.h:21]: Ok !

**Comment by EmilieNumworks** [comment]
Thanks @RedGl0w, this fixes an old bug!

**Pull request merged**